spec/activerecord-bitemporal/bitemporal_spec.rb

```diff
@@ -1773,4 +1773,118 @@ class EmployeeWithUniquness < Employee
       end
     end
   end
+
+  describe "#bitemporal_build_update_records" do
+    let(:from) { time_current }
+    let(:to) { from + 10.days }
+    let(:finish) { Time.utc(9999, 12, 31).in_time_zone }
+    let!(:employee) { Employee.create!(name: "Jone", valid_from: from, valid_to: to) }
+    let(:now) { time_current }
+    let(:force_update) { false }
+
+    subject do
+      employee.name = "Tom"
+      employee.__send__(:bitemporal_build_update_records, valid_datetime: now, current_time: now, force_update: force_update)
+    end
+
+    # before: |-----------------------|
+    # update:             *
+    # after:  |-----------|
+    #                     |-----------|
+    context "now time is in" do
+      let(:now) { from + 5.days }
+
+      it "returns the current valid record and before/after records" do
+        current, before, after = subject
+
+        aggregate_failures do
+          expect(current).to have_attributes(transaction_to: now, deleted_at: now)
+          expect(before).to have_attributes(name: "Jone", valid_from: from, valid_to: now, transaction_from: now)
+          expect(after).to have_attributes(name: "Tom", valid_from: now, valid_to: to, transaction_from: now)
+        end
+      end
+
+      context "with force_update" do
+        let(:force_update) { true }
+
+        it "returns the current valid record and an after record" do
+          current, before, after = subject
+
+          aggregate_failures do
+            expect(current).to have_attributes(transaction_to: now, deleted_at: now)
+            expect(before).to be_nil
+            expect(after).to have_attributes(name: "Tom", valid_from: from, valid_to: to, transaction_from: now)
+          end
+        end
+      end
+    end
+
+    # before:          |-----------|     |-----------|
+    # update:  *
+    # after:   |-------|
+    #                  |-----------|     |-----------|
+    context "now time is before" do
+      before do
+        Employee.create!(bitemporal_id: employee.bitemporal_id, valid_from: to + 5.days, valid_to: to + 10.days)
+      end
+      let(:now) { from - 5.days }
+
+      it "returns only an after record" do
+        current, before, after = subject
+
+        aggregate_failures do
+          expect(current).to be_nil
+          expect(before).to be_nil
+          expect(after).to have_attributes(name: "Tom", valid_from: now, valid_to: from, transaction_from: now)
+        end
+      end
+
+      context "with force_update" do
+        let(:force_update) { true }
+
+        it "returns the current valid record and an after record" do
+          current, before, after = subject
+
+          aggregate_failures do
+            expect(current).to have_attributes(transaction_to: now, deleted_at: now)
```

Comment:
This behavior looks weird... There are no valid records at that time, but the `current_valid_record` is not nil. This is because force update effectively ignores the passed `valid_datetime`.

Reply:
Thanks :)
I agree, I'll check it out later.